system/lib/wasmfs/syscalls.cpp

```diff
@@ -1006,10 +1006,15 @@ long __syscall_utimensat(int dirFD,
 
   // TODO: Set tv_nsec (nanoseconds) as well.
   // TODO: Handle tv_nsec being UTIME_NOW or UTIME_OMIT.
-  // TODO: Handle NULL times.
   // TODO: Check for write access to the file (see man page for specifics).
-  auto aSeconds = times[0].tv_sec;
-  auto mSeconds = times[1].tv_sec;
+  time_t aSeconds, mSeconds;
+  if (times == NULL) {
+    aSeconds = time(NULL);
+    mSeconds = aSeconds;
+  } else {
+    aSeconds = times[0].tv_sec;
+    mSeconds = times[1].tv_sec;
```

Comment:
I'm curious how / why this was needed?   Do we run test_utime under WASMFS?  Or did you try running manually with this flag?

Reply:
This was needed to implement ``utime(path, NULL)`` for wasmfs. The PR fixes ``utime(path, times)`` on noderawfs and implements ``utime(path, NULL)`` for all fs layers.

``test_utime`` was failing on ``test_core.wasmfs``, https://app.circleci.com/pipelines/github/emscripten-core/emscripten/19754/workflows/57871d6d-c5f8-4f0c-83ae-2566038f9724/jobs/513149

Reply:
At thanks.. i had forgotten that we can run the whole of the core tests suite in wasmfs mode